router/core/graph_server.go

```diff
@@ -1345,10 +1346,27 @@ func (s *graphServer) setupPluginHost(ctx context.Context, config *nodev1.Engine
 					return fmt.Errorf("failed to get plugin path: %w", err)
 				}
 
+				paramsMap := make(map[string]string)
+
+				if s.traceConfig.Enabled && len(s.traceConfig.Exporters) > 0 {
+					enabledExporters := make([]*rtrace.ExporterConfig, 0)
+					for _, exporter := range s.traceConfig.Exporters {
+						if !exporter.Disabled {
+							enabledExporters = append(enabledExporters, exporter)
+						}
+					}
+					exporterJson, err := json.Marshal(enabledExporters)
+					if err != nil {
+						return fmt.Errorf("failed to marshal exporters")
+					}
+					paramsMap["router_exporter_config"] = string(exporterJson)
```

Comment:
We pass in parts of the configuration of the router to the plugin as environment variables. This will include the graph token in case of cosmo cloud (and other tokens for other platforms).

Other options I saw were
* Pass them as a cmd parameter on startup like `plugin --tracing <values>`, this seems worse than the current approach
* Have a specific RPC method that gets called on startup, though this would mean that when generating the plugin we would need to have a standard startup method that can be called by the router.

Lmk your thoughts @Noroth @StarpTech

Reply:
Sounds good, also makes sense to have a config that we serialise which we can add to.